scripts/dotnet.py

```diff
@@ -181,10 +181,22 @@ def get_host_commit_sha(dotnet_path: str = None) -> str:
         # First look for the host information, since that is the sha we are
         # looking for. Then, grab the first Commit line we find, which will be
         # the sha of the framework we are testing
+        #
+        # Sample input for .NET Core 2.1+:
+        # Host (useful for support):
+        #   Version: 3.0.0-preview1-27018-05
+        #   Commit:  7a7ca06512
+        #
+        # Sample input for .NET Core 2.0:
+        # Product Information:
+        #   Version:            2.1.202
+        #   Commit SHA-1 hash:  281caedada
         if 'Host' in decoded_line:
             foundHost = True
+        elif 'Product' in decoded_line:
+            foundHost = True
         elif foundHost and 'Commit' in decoded_line:
-            return decoded_line.strip().split()[1]
+            return decoded_line.strip().split()[-1]
```

Comment:
[-1] is the last element. the string gets trimmed first, so this will always be our sha

Reply:
The difference between using element in position 1 vs using the last element is that the current dotnet cli have two tokens and we select the second one as being the SHA. In the future, if this changes, we do not know what is the last token being selected. I would suggest leaving it as-is.

Reply:
I think that using the last element is actually more future-proof

For .NET Core 2.0 we had:

```log
Commit SHA-1 hash:  281caedada
```

For .NET Core 2.1+ we have:

```log
Commit:  7a7ca06512
```

I think that if it's ever going to change, somebody is going to add text before the SHA, not after it. Also, now we have a CI that validates that, so if it ever breaks we will know about this with the first broken PR. Maestro sends few PRs a day so we are going to find this out very quickly ;)

Reply:
Maybe moving forward we should use a regex to match this information. Something along the lines of: `^\W+Commit:\W+\b[0-9a-f]{5,40}\b$`.

Reply:
that would be most probably the best solution!